contrib/mesos/pkg/minion/tasks/task.go

```diff
@@ -29,7 +29,10 @@ import (
 	"k8s.io/kubernetes/contrib/mesos/pkg/runtime"
 )
 
-const defaultTaskRestartDelay = 5 * time.Second
+const (
+	defaultTaskRestartDelay = 5 * time.Second
+	defaultKillGracePeriod  = 5 * time.Second
```

Comment:
you've struck upon an ugly side of mesos. there's no easy way for us to discover the grace period that we actually get from mesos: it's simple a missing core feature. there's a MESOS-xyz ticket for this somewhere.

if it was discoverable then we could come up with a better strategy. there are some comments in the executor regarding this as well (because there we're concerned about cleaning up pods within the grace period).